### PR TITLE
[Feat] Add PointCloud Segmentation

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds a PointCloudSegmentation task

Screenshot 2021-07-14 at 13 08 03

pointcloud_predictions

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/api/pointcloud.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+    :template: classtemplate.rst
+
+    ~segmentation.data.PointCloudSegmentationData
+    ~segmentation.data.PointCloudSegmentationPreprocess
+    ~segmentation.data.PointCloudSegmentationFoldersDataSource
+    ~segmentation.data.PointCloudSegmentationDatasetDataSource
+    ~segmentation.model.PointCloudSegmentation
```
Comment (ethanwharris) [inline docs/source/api/pointcloud.rst:24]:
```suggestion
    ~segmentation.model.PointCloudSegmentation
    ~segmentation.data.PointCloudSegmentationData

    segmentation.data.PointCloudSegmentationPreprocess
    segmentation.data.PointCloudSegmentationFoldersDataSource
    segmentation.data.PointCloudSegmentationDatasetDataSource
```

File: docs/source/reference/semantic_segmentation.rst [hunk 2]
```diff
@@ -34,9 +34,9 @@ Here's the structure:
         ├── F61-2.png
             ...
 
-Once we've downloaded the data using :func:`~flash.core.data.download_data`, we create the :class:`~flash.image.segmentation.data.SemanticSegmentationData`.
-We select a pre-trained ``mobilenet_v3_large`` backbone with an ``fpn`` head to use for our :class:`~flash.image.segmentation.model.SemanticSegmentation` task and fine-tune on the CARLA data.
-We then use the trained :class:`~flash.image.segmentation.model.SemanticSegmentation` for inference.
+Once we've downloaded the data using :func:`~flash.core.data.download_data`, we create the :class:`~flash.image.segmentation.data.PointCloudSegmentationData`.
+We select a pre-trained ``randlanet_PointCloud_kitti`` backbone for our :class:`~flash.image.segmentation.model.PointCloudSegmentation` task and fine-tune on the CARLA data.
+We then use the trained :class:`~flash.image.segmentation.model.PointCloudSegmentation` for inference.
```
Comment (ethanwharris) [inline docs/source/reference/semantic_segmentation.rst:39]: This looks like an accidental change

File: flash/pointcloud/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from flash.pointcloud.segmentation.data import PointCloudSegmentationData  # noqa: F401
+from flash.pointcloud.segmentation.data import PointCloudSegmentationDatasetDataSource  # noqa: F401
+from flash.pointcloud.segmentation.data import PointCloudSegmentationFoldersDataSource  # noqa: F401
+from flash.pointcloud.segmentation.data import PointCloudSegmentationPreprocess  # noqa: F401
```
Comment (ethanwharris) [inline flash/pointcloud/__init__.py:4]: We normally just do the model and the datamodule right?
```suggestion
```

File: flash_examples/pointcloud_segmentation.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "data/SemanticKittiTiny/predict/000000.bin",
+    "data/SemanticKittiTiny/predict/000001.bin",
+])
+
+# 5. Save the model!
+trainer.save_checkpoint("pointcloud_segmentation_model.pt")
+
+# 6. Visualize
+app = launch_app(datamodule)
+app.show_predictions(predictions)
```
Comment (ethanwharris) [inline flash_examples/pointcloud_segmentation.py:45]: We should move this to a `flash_examples/visualize` along with the fiftyone examples and one showing the data fetcher stuff. Can do this in a separate PR though

File: flash/pointcloud/segmentation/model.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        metrics: Any metrics to use with this :class:`~flash.core.model.Task`. If ``None``, a default will be selected
+            by the :class:`~flash.core.classification.ClassificationTask` depending on the ``multi_label`` argument.
+        learning_rate: The learning rate for the optimizer.
+        multi_label: If ``True``, this will be treated as a multi-label classification problem.
+        serializer: The :class:`~flash.core.data.process.Serializer` to use for prediction outputs.
+    """
+
+    backbones: FlashRegistry = POINTCLOUD_SEGMENTATION_BACKBONES
+
+    @requires_extras("pointcloud")
```
Comment (ethanwharris) [inline flash/pointcloud/segmentation/model.py:97]:
```suggestion

    required_extras: str = "pointcloud"

```

File: docs/source/api/pointcloud.rst (hunk 1 above)
Comment (tchaton) [reply docs/source/api/pointcloud.rst:24]: Done !

File: docs/source/reference/semantic_segmentation.rst (hunk 2 above)
Comment (tchaton) [reply docs/source/reference/semantic_segmentation.rst:39]: Removed back ! Good catch !

File: flash/pointcloud/__init__.py (hunk 3 above)
Comment (tchaton) [reply flash/pointcloud/__init__.py:4]: Done !

File: flash_examples/pointcloud_segmentation.py (hunk 4 above)
Comment (tchaton) [reply flash_examples/pointcloud_segmentation.py:45]: Worth to brainstorm a little bit about this one.

File: flash/pointcloud/segmentation/model.py (hunk 5 above)
Comment (tchaton) [reply flash/pointcloud/segmentation/model.py:97]: Done !

File: docs/source/api/pointcloud.rst (hunk 1 above)
Comment (ethanwharris) [inline docs/source/api/pointcloud.rst:24]: I think this got lost somewhere:

```suggestion
    ~segmentation.model.PointCloudSegmentation
    ~segmentation.data.PointCloudSegmentationData

    segmentation.data.PointCloudSegmentationPreprocess
    segmentation.data.PointCloudSegmentationFoldersDataSource
    segmentation.data.PointCloudSegmentationDatasetDataSource
```

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley:
